File: cloud-functions/submitTrade/trade/exchangeSession.js
const CustomError = require("../utils/error"); // Assuming CustomError is in this path

/**
 * Base class for implementing different exchange sessions.
 */
class ExchangeSession {
  /**
   * Create a new ExchangeSession instance.
   * @param {string} apiKey - The API key for the session.
   * @param {string} apiSecret - The API secret for the session.
   */
  constructor(apiKey, apiSecret) {
    this.apiKey = apiKey;
    this.apiSecret = apiSecret;
  }

  /** 
   * Execute a bulkOrder for the session
   * @async
   * @throws {CustomError} Thorws a custom error if method not implemented by subclass.
   */
  async bulkOrder(payload) {
    throw new CustomError({
      message: "Method bulkOrder must be implemented by subclass",
      status: 501,
      source: "bulkOrder",
    });
  }
}

module.exports = ExchangeSession;

File: cloud-functions/submitTrade/trade/exchange_session.py
class ExchangeSession:
    """
    Base class for implementing different exchange sessions.
    """
    
    def __init__(self, api_key, api_secret):
        """
        Create a new ExchangeSession instance.
        
        :param api_key: The API key for the session.
        :param api_secret: The API secret for the session.
        """
        self.api_key = api_key
        self.api_secret = api_secret